module/zfs/dbuf.c

```diff
@@ -4460,7 +4521,8 @@ dbuf_write_override_done(zio_t *zio)
 	if (!BP_EQUAL(zio->io_bp, obp)) {
 		if (!BP_IS_HOLE(obp))
 			dsl_free(spa_get_dsl(zio->io_spa), zio->io_txg, obp);
-		arc_release(dr->dt.dl.dr_data, db);
```

Comment:
Elsewhere you check that `dr_data != NULL && dr_data != db->db_buf` Is there reason you only need to check for NULL here? If so please comment why.

Reply:
See my comment above. We will clean this up as well when fixing the other dbuf issues.